Meadowlark-js/packages/meadowlark-authz-server/src/handler/ResetClientSecret.ts

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import crypto from 'node:crypto';
+import { authorizationHeader, writeErrorToLog } from '@edfi/meadowlark-core';
+import { writeDebugStatusToLog, writeRequestToLog } from '../Logger';
+import { ensurePluginsLoaded, getAuthorizationStore } from '../plugin/AuthorizationPluginLoader';
+import { checkForAuthorizationErrors } from '../security/JwtValidator';
+import { clientIdFrom } from '../Utility';
+import { AuthorizationRequest } from './AuthorizationRequest';
+import { AuthorizationResponse } from './AuthorizationResponse';
+import { ResetClientSecretResponseBody } from '../model/ResetClientSecretResponseBody';
+import { ResetAuthorizationClientSecretResult } from '../message/ResetAuthorizationClientSecretResult';
+import { ResetAuthorizationClientSecretRequest } from '../message/ResetAuthorizationClientSecretRequest';
+
+const moduleName = 'handler.UpdateClientSecret';
+
+export async function resetAuthorizationClientSecret(
+  authorizationRequest: AuthorizationRequest,
+): Promise<AuthorizationResponse> {
+  try {
+    writeRequestToLog(moduleName, authorizationRequest, 'createClient');
+    await ensurePluginsLoaded();
+
+    const errorResponse: AuthorizationResponse | undefined = checkForAuthorizationErrors(
+      authorizationHeader(authorizationRequest.headers),
+    );
+
+    if (errorResponse != null) {
+      writeDebugStatusToLog(moduleName, authorizationRequest, 'createClient', errorResponse.statusCode, errorResponse.body);
+      return errorResponse;
+    }
+
+    const clientSecretBytes: Buffer = crypto.randomBytes(32);
+    const clientSecretHashed: string = crypto.createHash('shake256').update(clientSecretBytes).digest('hex');
```

Comment:
These lines are duplicated from somewhere else, right? If so, can we move to a shared function so that changes to the random secret algorithm occur in only one place?

Reply:
@stephenfuqua yes - in the createClient. Brad had a function that did the work of the createHash. I've updated to follow the same convention in this code